### PR TITLE
[#174416949] Replace TSLint with ESLint

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes
- Replace `tslint` and `italia-tslint-rules` with `@pagopa/eslint-config`
- Add eslint configuration from `@pagopa/eslint-config/strong`
- Replace `tslint:disable` with `eslint-disable`
- Change `lint` script inside `package.json`

#### Motivation and Context
TSLint has been deprecated in favour of ESLint. This PR replace TSLint with ESLint and `italia-tslint-rules` with the new package `@pagopa/eslint-config`. Some rules from `@pagopa/eslint-config/strong` has been disabled to minimize code refactoring.

#### How Has This Been Tested?
- Local `yarn lint` and `yarn build` execution

#### Types of changes

- [X] Chore
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
+  "env": {
+      "es6": true,
+      "node": true
+  },
+  "ignorePatterns": [
```
Comment (gunzip) [inline .eslintrc.js:6]: why do we need both this and .eslintignore?

**Review by gunzip — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (BurnedMarshal) [reply .eslintrc.js:6]: Good point. We don't. Fixed in ffcd79b

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      "node_modules",
+      "generated",
+      "**/__tests__/*",
+      "**/__mocks__/*",
+      "Dangerfile.*",
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply .eslintrc.js]: You're right. `Dangerfile.ts` is now under lint with afbe0b1

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+node_modules
+
+# We shouldn't lint assets nor generated files
+generated
+
+**/__tests__/*
+**/__mocks__/*
+Dangerfile.*
+jest.config.js
+.eslintrc.js
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply .eslintignore]: The file was removed in 4e899d2

File: .eslintrc.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (balanza) [reply .eslintrc.js]: well I meant... why not tests and mocks as well?

**Pull request merged**